packages/@aws-cdk/aws-iotevents/lib/detector-model.ts

```diff
@@ -0,0 +1,46 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnDetectorModel } from './iotevents.generated';
+import { State } from './state';
+
+/**
+ * Properties for defining an AWS IoT Events detector model
+ */
+export interface DetectorModelProps {
+  /**
+   * The name of the detector model
+   *
+   * @default - CloudFormation will generate a unique name of the detector model
+   */
+  readonly detectorModelName?: string,
+
+  /**
+   * The state that is entered at the creation of each detector.
+   */
+  readonly initialState: State;
+}
+
```

Comment:
Feels like we're missing an `IDetectorModel` interface with the `readonly detectorModelName: string` property.